### ISSUE TITLE
undefined reference to 'ethash_cuda_miner::c_defaultGridSize'

### ISSUE THREAD

**Issue by soulmachine** [Genoil/cpp-ethereum#102]
I'm building the branch `110` on Ubuntu 16.04 with Cuda 8.0, I get the following errors:

```
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'ethash_cuda_miner::c_defaultGridSize'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'ethash_cuda_miner::c_defaultBlockSize'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'ethash_cuda_miner::c_defaultNumStreams'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::g_logVerbosity'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashGPUMiner::s_devices'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashGPUMiner::s_devices'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashGPUMiner::s_devices'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashGPUMiner::s_devices'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashGPUMiner::configureGPU(unsigned int, unsigned int, unsigned int, unsigned int, bool, unsigned int, unsigned long, unsigned int, unsigned int)'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashGPUMiner::getNumDevices()'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashGPUMiner::s_numInstances'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'EthStratumClientV2::EthStratumClientV2(dev::eth::GenericFarm<dev::eth::EthashProofOfWork>*, dev::eth::MinerType, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int const&, int const&, int const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'EthStratumClientV2::setFailover(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'EthStratumClientV2::~EthStratumClientV2()'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::LogOutputStreamBase::LogOutputStreamBase(char const*, std::type_info const*, unsigned int, bool)'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::LogOutputStreamBase::LogOutputStreamBase(char const*, std::type_info const*, unsigned int, bool)'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashCUDAMiner::s_devices'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashCUDAMiner::s_devices'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashCUDAMiner::s_devices'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashCUDAMiner::s_devices'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashCUDAMiner::getNumDevices()'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashCUDAMiner::s_numInstances'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashCUDAMiner::configureGPU(unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned long, unsigned int, unsigned int)'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'EthStratumClientV2::~EthStratumClientV2()'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashCUDAMiner::listDevices()'
/tmp/cct6U8kh.ltrans0.ltrans.o:<artificial>:function main: error: undefined reference to 'dev::eth::EthashGPUMiner::listDevices()'
```

Any idea ? Thanks

**Comment by Darkness4** [comment]
I got this error too. I have also CUDA 8.0 with Ubuntu 14.04. I used -DCOMPUTE=52 because i have a gtx970 and -DBUNDLE=cudaminer.  Branch 110 only, master is fine. 
I compiled with the latest cmake (3.6.2), gcc-5 and g++-5 (Ubuntu 5.4.1-2ubuntu1~14.04). 
Same error with gcc-4.9 and g++-4.9 (Ubuntu 4.9.4-2ubuntu1~14.04.1)
gcc-6 is incompatible. 

With gcc-4.8 (Ubuntu 4.8.5-2ubuntu1~14.04.1) 4.8.5, it gave only 

`"/tmp/ccdNojto.ltrans25.ltrans.o:<artificial>:_ZL15__fatDeviceText.293534.5840: error: undefined reference to 'fatbinData'"
`

It always blocks at 

`[100%] Linking CXX executable ethminer`
